portfolio/src/main/webapp/script.js

```diff
@@ -17,12 +17,81 @@
  */
 function addRandomFact() {
   const facts =
-      ['I have a pet dog', 'I like to code'];
+      ['I have a pet dog!', 'I like to code!', 'Marquees are my favourite HTML element!'];
 
   // Pick a random fact.
   const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
+}
+
+/**
+ * Controls slideshow image display on the page
+ */
+const deckNums = { // Enumerate slide decks
+  DOG_DECK: 0,
+  PLACE_DECK: 1,
+  FOOD_DECK: 2,
+};
+
+const PREV_SLIDE_DIR = -1;
+const NEXT_SLIDE_DIR = 1;
+
+class Slideshow {
+  constructor(idNum, slides, dom) {
+    this.index = 0;
+    this.idNum = idNum;
+    this.slides = slides;
+    this.dom = dom;
```

Comment:
Instead of passing in the img tag, what if we passed in the slideshow div that the img is contained in?

Then we could set the event handlers inside the constructor, thus encapsulating all slideshow logic inside this class.

If you want to go down that route you may want to use `querySelectorAll` [1]

[1] https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/querySelectorAll